### PR TITLE
(#178) Fix structured output JSON descriptions

### LINKED ISSUE #178: LLMDescription not included in the generated schema
It seems to me like the `JsonStructuredData` schema doesn't actually contain the annotated data from `@LLMDescription`

For example this is what the result is when serializing an annotated class: 

```
    @Serializable
    @LLMDescription("A simple data class for testing serialization")
    data class Hero(val name: String, val age: Int, @LLMDescription("The hero's superpower") val superpower: String)

    test("Serializing simple data") {
      val heroStructure =
        JsonStructuredData.createJsonStructure<Hero>(
          schemaFormat = JsonSchemaGenerator.SchemaFormat.Simple,
          schemaType = JsonStructuredData.JsonSchemaType.SIMPLE,
        )

      expectSelfie(serialize.encodeToString(heroStructure.schema))
        .toBe(
          """{
    "type": "ai.koog.prompt.params.LLMParams.Schema.JSON.Simple",
    "name": "Hero",
    "schema": {
        "type": "object",
        "properties": {
            "name": {
                "type": "string"
            },
            "age": {
                "type": "integer"
            },
            "superpower": {
                "type": "string"
            }
        },
        "required": [
            "name",
            "age",
            "superpower"
        ]
    }
}"""
        )
    }
  })
```

I would have expected the schema to be this:

```
{
    "type": "ai.koog.prompt.params.LLMParams.Schema.JSON.Simple",
    "name": "Hero",
    "description": "A simple data class for testing serialization", <----
    "schema": {
        "type": "object",
        "properties": {
            "name": {
                "type": "string"
            },
            "age": {
                "type": "integer"
            },
            "superpower": {
                "type": "string",
                "description": "The hero's superpower" <----
            }
        },
        "required": [
            "name",
            "age",
            "superpower"
        ]
    }
}
```

### PR DESCRIPTION
**Pull request by EugeneTheDev**
Based on #188 and fixes #178 by properly handling `@LLMDescription` during JSON schema generation. Also updated the tests to include the annotation too.

### CODE REVIEW

File: prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt [hunk 1]
```diff
@@ -67,26 +66,29 @@ public class JsonStructuredData<TStruct>(
...
         /**
          * Factory method to create JSON structure with auto-generated JSON schema.
+         *
+         * @param id Unique identifier for the structure.
+         * @param serializer Serializer used for converting the data to and from JSON.
+         * @param json JSON configuration instance used for serialization.
+         * @param schemaFormat Format of the generated schema, can be simple or detailed.
+         * @param maxDepth Maximum recursion depth when generating schema to prevent infinite recursion for circular references.
+         * @param descriptions Optional map of serial class names and property names to descriptions.
+         * If a property/type is already described with [LLMDescription] annotation, value from the map will override this description.
```
Comment (Ololoshechkin) [inline prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt:128]: Let's also please add good example in KDoc about how to form the descriptions map.
I mean, `"className.propertyName" to "property description"`

**Review by Ololoshechkin — APPROVED** [review]
@EugeneTheDev please also add a test with a lot of different descriptors:
- class descriptor via annotation
- property descriptor via annotation
- property of Serializable type, also with field, each of them annotated
- one field is annotated and also overriden by the map
- one field is NOT annotated but provided through the map
- what else can be an edge case?

File: prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt (hunk 1 above)
Comment (Ololoshechkin) [reply prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt:128]: Also, maybe let's name it `descriptionOverrides` if it's an overriding map?

Comment (EugeneTheDev) [reply prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt:128]: Regarding testing edge cases, that's what I already modified in tests, to cover these cases you described. I didn't mark each field with annotation though because it looks redundant, and it's also nice to test that description is added only for these fields where it is specified, and is skipped for other fields.

Regarding examples and parameter renaming: sounds good, will do it

**Review by Rizzen — APPROVED** [review]
